framework/src/base/NonlinearSystem.C

```diff
@@ -680,16 +694,28 @@ NonlinearSystem::computeResidual(NumericVector<Number> & residual, Moose::Kernel
 
   try
   {
-    residual.zero();
```

Comment:
That is true. If we are not evaluating `TimeKernel`, logically we should not call anything in time integrator, right? Residual is not zeroed out at the beginning because it only has contribution from nontime residual. You can see at the end there is a direct assignment to this residual vector.